domain/src/main/kotlin/domain/actions/AddThingAction.kt

```diff
@@ -23,11 +23,11 @@ class AddThingAction(
     private val idOnDeviceRetriever: IdOnDeviceRetriever
 ) {
 
-    fun add(deviceId: DeviceId, addThingRequest: AddThingRequest): Either<ActionError, Unit> =
-        deviceRepository.retrieve(deviceId).fold(
+    fun add(addThingRequest: AddThingRequest): Either<ActionError, Unit> =
+        deviceRepository.retrieve(addThingRequest.deviceId).fold(
```

Comment:
MAJOR: this stuff (the Action I mean, not necessarily the Repository) must returns the added Thing